Source/Core/InputCommon/InputProfile.h

```diff
@@ -1,53 +1,30 @@
-// Copyright 2018 Dolphin Emulator Project
+// Copyright 2019 Dolphin Emulator Project
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
 #pragma once
 
-class InputConfig;
+#include "Common/IniFile.h"
+
+#include <string>
 
 namespace ControllerEmu
 {
 class EmulatedController;
 }
 
-#include <string>
-#include <vector>
-
-namespace InputProfile
+namespace InputCommon
 {
-std::vector<std::string> GetProfilesFromSetting(const std::string& setting,
-                                                const std::string& root);
-
-enum class CycleDirection : int
-{
-  Forward = 1,
-  Backward = -1
-};
-
-class ProfileCycler
+class InputProfile
 {
 public:
-  void NextWiimoteProfile(int controller_index);
-  void PreviousWiimoteProfile(int controller_index);
-  void NextWiimoteProfileForGame(int controller_index);
-  void PreviousWiimoteProfileForGame(int controller_index);
+  explicit InputProfile(const std::string& profile_path);
```

Comment:
This should probably have the destructor defaulted in the cpp file, given the class contains entirely non-trivial types.